go-controller/pkg/clustermanager/userdefinednetwork/controller.go

```diff
@@ -358,3 +358,60 @@ func updateCondition(conditions []metav1.Condition, cond *metav1.Condition) ([]m
 	}
 	return conditions, false
 }
+
+// UpdateSubsystemError may be used by other controllers handling UDN/NAD/network setup to report any problems that
+// may affect UDN functionality.
+// FieldManager should be unique for every subsystem. When errorMessage is empty, condition will be removed.
+// If given network is not managed by a UDN, no error will be returned.
+func (c *Controller) UpdateSubsystemError(networkName string, errorMessage string, fieldManager string) error {
+	// try to find udn using network name
+	udnNamespace, udnName := template.ParseNetworkName(networkName)
+	if udnName == "" {
+		return nil
+	}
+	udn, err := c.udnClient.K8sV1().UserDefinedNetworks(udnNamespace).Get(context.Background(), udnName, metav1.GetOptions{})
+	if err != nil {
+		return nil
+	}
+	udnStatus := udnapplyconfkv1.UserDefinedNetworkStatus()
+	conditionType := fmt.Sprintf("SubsystemError.%s", fieldManager)
```

Comment:
Currently we report zone status with conditions with type `Ready-In-Zone+<zoneName>` and I think this is the only condition we have for now.
It looks like we have 2 condition types for now: readiness and errors. Readiness is reported always and may contain error message if status if false. We can use those for per-zone readiness reports and for subsystem reports (if needed). Error condition is only reported when an error occurred and is not added otherwise. I think Error conditions make sense in case we have many subsystems reporting errors in a status and we don't want to overload the status with "no error" messages.
If that makes sense, every subsystem should only report one type of condition: readiness or error. In that case
`<fieldmanager>.k8s.ovn.org/<type>` works. But if we ever get another condition type or want the same fieldManager to report 2 types, we may need to switch to `type=<subsystem>.k8s.ovn.org/<type>` + `fieldmanager = <subsystem.type>` in case subsystem wants to update those conditions separately with different filed managers. Probably too much for now, I can go with single-type assumption

Reply:
Not sure I get you. If we want the same fieldManager to report 2 types, whats wrong with this scheme:
```
fieldmanager1.k8s.ovn.org/type1
fieldmanager1.k8s.ovn.org/type2
```
I am assuming `subsystem == field manager`

Is the main cluster manager UDN reconciler not reporting any condition already?